### PR TITLE
Change namespaces from onemkl to oneapi::mkl and header structure

### PR DESCRIPTION
**Pull request by vmalia**
# Description

*Updated namespaces from onemkl to oneapi::mkl
*Updated calls to mkl::gpu backend to avoid namespace conflict with oneapi::mkl
*Updated directory structure in include/
*Renamed onemkl.hpp to mkl.hpp
*Updated README.md with above changes
*Updated Contributing guide with above changes
*Updated documentation with above changes

### CODE REVIEW

File: include/oneapi/mkl/blas/detail/cublas/blas_ct.hpp [hunk 1]
```diff
@@ -47,7 +48,7 @@ void syr2<library::cublas, backend::nvidiagpu>(cl::sycl::queue &queue, uplo uppe
                                                cl::sycl::buffer<float, 1> &y, std::int64_t incy,
                                                cl::sycl::buffer<float, 1> &a, std::int64_t lda) {
     syr2_precondition(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
-    onemkl::cublas::syr2(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
+    oneapi::mkl::cublas::syr2(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
```
Comment (mmeterel) [inline include/oneapi/mkl/blas/detail/cublas/blas_ct.hpp:51]: Not related to this MR, but in general, maybe cublas should not be inside mkl namespace, especially now that we have a new namespace called oneapi.

File: include/oneapi/mkl/blas/detail/mklcpu/blas_ct.hpp [hunk 2]
```diff
@@ -47,7 +48,7 @@ void syr2<library::intelmkl, backend::intelcpu>(cl::sycl::queue &queue, uplo upp
                                                 cl::sycl::buffer<float, 1> &y, std::int64_t incy,
                                                 cl::sycl::buffer<float, 1> &a, std::int64_t lda) {
     syr2_precondition(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
-    onemkl::mklcpu::syr2(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
+    oneapi::mkl::mklcpu::syr2(queue, upper_lower, n, alpha, x, incx, y, incy, a, lda);
```
Comment (mmeterel) [inline include/oneapi/mkl/blas/detail/mklcpu/blas_ct.hpp:51]: I think, the new direction is to use external/public APIs from Intel oneMKL. In that case, we should think how to handle mklcpu/mklgpu namespaces.

File: src/blas/backends/mklgpu/mkl_internal_blas_gpu_wrappers.cpp [hunk 3]
```diff
@@ -24,1143 +24,1166 @@
...
 // Buffer APIs
 
-void gemm(cl::sycl::queue &queue, onemkl::transpose transa, onemkl::transpose transb,
+void gemm(cl::sycl::queue &queue, oneapi::mkl::transpose transa, oneapi::mkl::transpose transb,
           std::int64_t m, std::int64_t n, std::int64_t k, float alpha,
           cl::sycl::buffer<float, 1> &a, std::int64_t lda, cl::sycl::buffer<float, 1> &b,
           std::int64_t ldb, float beta, cl::sycl::buffer<float, 1> &c, std::int64_t ldc) {
-    mkl::gpu::sgemm(queue, mkl::cblas_convert(transa), mkl::cblas_convert(transb), m, n, k, alpha,
-                    a, lda, b, ldb, beta, c, ldc);
+    ::mkl::gpu::sgemm(queue, ::mkl::cblas_convert(transa), ::mkl::cblas_convert(transb), m, n, k,
```
Comment (mmeterel) [inline src/blas/backends/mklgpu/mkl_internal_blas_gpu_wrappers.cpp:38]: The namespaces will be updated when beta09 is available, right? We will also change open source to use public APIs at some point.

File: include/oneapi/mkl/blas/detail/cublas/blas_ct.hpp (hunk 1 above)
Comment (vmalia) [reply include/oneapi/mkl/blas/detail/cublas/blas_ct.hpp:51]: Let's bring this up as a proposal later.

File: include/oneapi/mkl/blas/detail/mklcpu/blas_ct.hpp (hunk 2 above)
Comment (vmalia) [reply include/oneapi/mkl/blas/detail/mklcpu/blas_ct.hpp:51]: Let's bring this up in a proposal later.

File: src/blas/backends/mklgpu/mkl_internal_blas_gpu_wrappers.cpp (hunk 3 above)
Comment (vmalia) [reply src/blas/backends/mklgpu/mkl_internal_blas_gpu_wrappers.cpp:38]: Another change to namespaces will come soon.

**Pull request merged**